app/model/editions/templates/Capi.scala

```diff
@@ -0,0 +1,65 @@
+package model.editions.templates
+
+import scala.language.postfixOps
+
+case class Capi(val string: String, val next: Option[Capi] = None) {
+  override def toString = string + (if (next.isDefined) next.get.toString else "")
+  private def add(string: String): Capi = add(Capi(string))
+  private def add(newNext: Capi): Capi= {
+    if (!next.isDefined)
+      this.copy(next = Some(newNext))
+    else
+      this.copy(next = next map(_ add newNext))
+  }
+  def and(c:Capi) = c match {
+    case Capi(_, None) => this.add(Capi(",")).add(c)
+    case Capi(_, Some(c2)) => this.add(Capi(",")) .add("(").add(c).add(")")
+  }
+  def andnot(c:Capi) = c match {
+    case Capi(_, None) => this.add(Capi(",-")).add(c)
+    case Capi(_, Some(c2)) => this.add(Capi(",-")) .add("(").add(c).add(")")
+  }
+  def or(c:Capi) = add(Capi("|")).add(c)
+  def tag(s:String) = Capi(s)
+  def build(c:Capi) = this.copy(next = Some(c))
+}
+
+object Capi{
+  def apply() = new Capi("?tag=")
+  def apply(s: String) = new Capi(s)
+  def article=Capi("type/article")
+  def lifeandstyle=Capi("lifeandstyle/lifeandstyle")
+
+  def features=Capi("tone/features")
```

Comment:
I was wondering how do you reify the tags, or sections, etc. I see you are mapping them manually. But there are more than 50,000 tags. And they are mutating all the time, I guess? At least more are added frequently. So how would you handle that?

Reply:
I must say I really like it, it is very elegant and safe to be able to write queries like this. I am just not sure about this practical issue.

Reply:
Whoever is writing the client will define the tags they want.  This section of code would not be in a real client.

Reply:
The `import model.editions.templates.Capi._` import at the top where the items are used is all that matters.

Reply:
OK, so the client needs to define at compile time which tags they will use, which will work well if they use a limited and known set of them, for what I understand.

Reply:
You could easily make it dynamic.  Parse values out, wrap them into Capi class instances and include them.  That's what functional languages do best.